### ISSUE TITLE
Config reload on backend redeploy often leads to invalid state

### ISSUE THREAD

**Issue by Levino** [tutumcloud/haproxy#144]
I have haproxy running as a loadbalancer for backend services. The backend services are auto redeployed on registry push. In many cases the load balancer will give a 503 for the backend services, after they have been restarted. I suspect this is due to the fact that there are two different services running with the same docker image and these two are restarted at the same time.

I would say that the fix that "solved" this problemhttps://github.com/tutumcloud/haproxy/issues/95 is just of a temporary natures.

To recap: A live config reload of haproxy is kind of blocking. This should be fixed asap. It needs to be nonblocking and queue updates and process them in order or something.

**Comment by tifayuki** [comment]
Which version of tutum/haproxy are you using? The code that blocks haproxy from reloading is removed: https://github.com/tutumcloud/haproxy/commit/72b46679b697dbddaf036257cc76d6372bb3f4c1

Can you paste the logs when the blocking happened?

**Comment by lazabogdan** [comment]
Same issue here, running an AWS cluster on tutum. After a push/redeploy, I have to restart my haproxy services, otherwise (sometimes) I start getting 503s. Using the latest version

**Comment by razitz** [comment]
+1

**Comment by Levino** [comment]
I assume this is a race condition on many updates of the config from concurrently starting containers. Needs to be fixed asap. We had to switch off auto redeploy for production environment. Very sad!
